modin/numpy/arr.py

```diff
@@ -1310,72 +1416,93 @@ def sum(
         if self._ndim == 1:
             if axis == 1:
                 raise numpy.AxisError(1, 1)
-            result = self._query_compiler.astype(
-                {col_name: out_dtype for col_name in self._query_compiler.columns}
+            result = self._query_compiler
+            if isinstance(where, array):
+                result = where.where(self, 0)._query_compiler
+            result = result.astype(
+                {col_name: out_dtype for col_name in result.columns}
             ).sum(axis=0)
-            if initial is not None:
-                result = result.add(initial)
+            result = result.add(initial)
             if keepdims:
-                if initial is not None:
-                    if out is not None:
-                        out._query_compiler = (
-                            numpy.ones_like(out, dtype=out_dtype) * initial
-                        )._query_compiler
-                    else:
-                        out = array([initial], dtype=out_dtype)
+                if out is not None:
+                    out._query_compiler = (
+                        (numpy.ones_like(out, dtype=out_dtype) * initial)
+                        .astype(out_dtype)
+                        ._query_compiler
+                    )
                 if out is not None and out.shape != (1,):
                     raise ValueError(
                         f"operand was set up as a reduction along axis 0, but the length of the axis is {out.shape[0]} (it has to be 1)"
                     )
-                return fix_dtypes_and_determine_return(result, 1, dtype, out, where)
-            return result.to_numpy()[0, 0] if where else initial
+                return (
+                    fix_dtypes_and_determine_return(
+                        result, 1, dtype, out, where is not False
+                    )
+                    if where is not False or out is not None
+                    else array([initial], dtype=out_dtype)
+                )
+            return result.to_numpy()[0, 0] if where is not False else initial
         if axis is None:
+            result = self
+            if isinstance(where, array):
+                result = where.where(self, 0)
             result = (
-                self.flatten()
-                .astype(out_dtype)
-                .sum(
-                    axis=axis,
-                    dtype=dtype,
-                    out=out,
-                    keepdims=None,
-                    initial=initial,
-                    where=where,
-                )
+                result.astype(out_dtype)
+                ._query_compiler.sum(axis=1)
+                .sum(axis=0)
+                .to_numpy()
+                .flatten()[0]
```

Comment:
why do we need to `.flatten()`, cannot we just `[0, 0]` here? it should be reduced to a single cell anyway